extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -100,17 +119,37 @@ export abstract class AzureAuth implements vscode.Disposable {
 			if (!this.metadata.settings.microsoftResource) {
 				throw new Error(localize('noMicrosoftResource', "Provider '{0}' does not have a Microsoft resource endpoint defined.", this.metadata.displayName));
 			}
-			const result = await this.login(this.commonTenant, this.metadata.settings.microsoftResource);
-			loginComplete = result.authComplete;
-			if (!result?.response) {
-				Logger.error('Authentication failed');
-				return {
-					canceled: false
+			if (this._authLibrary === 'ADAL') {
+				const result = await this.login(this.commonTenant, this.metadata.settings.microsoftResource);
+				loginComplete = result.authComplete;
+				if (!result?.response) {
+					Logger.error('Authentication failed');
+					return {
+						canceled: false
+					};
+				}
+				const account = await this.hydrateAccount(result.response.accessToken, result.response.tokenClaims);
+				loginComplete?.resolve();
+				return account;
+			} else {
+				const result = await this.loginMsal(this.organizationsTenant, this.metadata.settings.microsoftResource);
```

Comment:
might make sense to change the MSAL them to login and the ADAL method to loginADAL, since we'll be deleting the ADAL code at some point (avoiding need to remember to cleanup later)

Reply:
I think I'll leave it as is for now, unless you feel very strongly about changing it to be loginADAL